source/operators/captions/webcaptions.py

```diff
@@ -0,0 +1,1016 @@
+
+    webcaptions_object.PutMediaCollection(operator_object.name, data)
+
+    operator_object.update_workflow_status("Complete")
+    return operator_object.return_output_object()
+
+
+# Create a VTT captions file from the web captions
+def create_vtt(event, context):
+    print("We got the following event:\n", event)
+
+    operator_object = MediaInsightsOperationHelper(event)
+    webcaptions_object = WebCaptions(operator_object)
+
+    try:
+        targetLanguageCodes = webcaptions_object.operator_object.configuration["TargetLanguageCodes"]
+    except KeyError as e:
+        webcaptions_object.operator_object.update_workflow_status("Error")
+        operator_object.add_workflow_metadata(WebCaptionsError="Missing a required metadata key {e}".format(e=e))
+        raise MasExecutionError(operator_object.return_output_object())
+
+    captions_collection = []
+    for lang in targetLanguageCodes:
+        webcaptions = []
+
+        webcaptions = webcaptions_object.GetWebCaptions(lang)
+
+        #captions = get_webcaptions_json(self.operator_object, lang)
+
+        vtt = webcaptions_object.WebCaptionsToVTT(webcaptions)
+        metadata = webcaptions_object.PutVTT(lang, vtt)
+
+        captions_collection.append(metadata)
+
+    data = {}
+    data["CaptionsCollection"] = captions_collection
+
+    webcaptions_object.PutMediaCollection(operator_object.name, data)
+
+    operator_object.update_workflow_status("Complete")
+    return operator_object.return_output_object()
+
+def start_translate_webcaptions(event, context):
+    print("We got the following event:\n", event)
+
+    operator_object = MediaInsightsOperationHelper(event)
+    webcaptions_object = WebCaptions(operator_object)
+
+    try:
+        source_lang = operator_object.configuration["SourceLanguageCode"]
+        target_langs = operator_object.configuration["TargetLanguageCodes"]
+    except KeyError:
+        operator_object.update_workflow_status("Error")
+        operator_object.add_workflow_metadata(TranslateError="Language codes are not defined")
+        raise MasExecutionError(operator_object.return_output_object())
+    try: 
+        terminology_names = operator_object.configuration["TerminologyNames"]
+    except KeyError:
+        terminology_names = []
+    try: 
+        parallel_data_names = operator_object.configuration["ParallelDataNames"]
+    except KeyError:
+        parallel_data_names = []
+
+    #webcaptions = get_webcaptions(operator_object, source_lang)
+    webcaptions = webcaptions_object.GetWebCaptions(source_lang)
+
+    # Translate takes a list of target languages, but it only allow one item in the list.  Too bad
+    # life would be so much easier if it truely allowed many targets.
+    webcaptions_object.TranslateWebCaptions(webcaptions, source_lang, target_langs, terminology_names, parallel_data_names)
+
+    return operator_object.return_output_object()
+
+
+def check_translate_webcaptions(event, context):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function calls 26 other functions. By comparison, 99% of the functions in the CodeGuru reference dataset use fewer. This indicates the function is highly coupled with other functions. A function that is highly coupled with other functions is difficult to understand and its behavior might change unexpectedly when one of its referenced functions is updated. High coupling could also increase the integration test complexity, maintenance cost and technical debt.

We recommend that you simplify this function or break it into multiple functions. For example, consider extracting the code blocks on lines 650-685, lines 691-764 into separate functions.